### ISSUE TITLE
Adding tag per checkpoint/model

### ISSUE THREAD

**Issue by Adil3tr** [adieyal/sd-dynamic-prompts#188]
A function that would be extremely useful would be if I could write *token* and then each generation in the batch would choose a random model and swap in the correct token phrase. And if it worked in reverse, where if I make an XY chart with multiple checkpoints on one axis, it'll swap in the right token for each one.

**Comment by adieyal** [comment]
Thanks for the suggestion. Not sure it's in scope for dynamic prompts, but I'll give it some thought. Wouldn't it run extremely slowly? Changing checkpoints can take a few seconds on fast hardware, and many seconds on slow drives.

**Comment by Adil3tr** [comment]
I regularly run XY grids with dozens of checkpoints, it only takes a few seconds for most of them to load into RAM. If the prompts were calculated beforehand, it could sort them into chunks by checkpoint and generate them with a minimum of switching. The XY charts alone would be really helpful because there's no way right now to generate a prompt across multiple checkpoints with the token changing in step to work.

**Comment by adieyal** [comment]
I see the usefulness of your suggestion - I'm not sure it fits within the scope though since I've tried to avoid adding functionality that manipulates settings. Let me take a look at XY Plot and see whether it's possible to do something there.

**Comment by Adil3tr** [comment]
Yeah I guess the alternative would be adding the ability for "__token__" to always be replaced with the token of the model currently loaded, with changing the model left to other scripts/the user.

**Comment by Bonedragon1300** [comment]
I would second something like this if it were possible, though with a way to specify a model, not just full randomize. maybe by putting a specific character/symbol on either side of the name of the model. maybe something like `<<"model name">>` to trigger it. could be useful for something like a character wildcard to switch to character specific models automatically

**Comment by Adil3tr** [comment]
> I would second something like this if it were possible, though with a way to specify a model, not just full randomize. maybe by putting a specific character/symbol on either side of the name of the model. maybe something like `<<"model name">>` to trigger it. could be useful for something like a character wildcard to switch to character specific models automatically

Fantastic point, there's no way to dynamically use subject models without this.
